### ISSUE TITLE
GlycineMine unfriendliness to other mines

### ISSUE THREAD

**Issue by adf-ncgr** [legumeinfo/mine-issues#173]
per @maxglycine, GlycineMine is not returning results for "links to other mines" on the gene report pages of other mines and is not receiving "links to other mines" from any of the others. This likely is some sort of configuration issue.

**Comment by adf-ncgr** [comment]
actually, that change has made glycinemine friendlier to other mines, but other mines seem to still be unfriendly to it. The browser console has a variety of errors, which probably suggest what else remains to be done to fix this.

**Comment by svengato** [comment]
Still trying to figure out what you changed, some of the libraries have changed but there must be source file(s) as well? And this was only in production?

**Comment by adf-ncgr** [comment]
see the commit linked above; it's in the github but only deployed on production at the moment.

**Comment by svengato** [comment]
That only changed a comment, so would anything really have changed?

**Comment by adf-ncgr** [comment]
the line below that was also different in the file. Maybe it had already been changed in the github, I didn't think to do a git pull, assuming the local files were already up-to-date

**Comment by svengato** [comment]
Is this correct?
```
friendlymines.linkgenerator = org.ncgr.intermine.bio.web.displayer.FriendlyMineLinkGenerator
```

**Comment by adf-ncgr** [comment]
I think so, I just copied the value another properly working mine had there.

**Comment by adf-ncgr** [comment]
actually, I am pretty confused by the current behaviors. AFAICT the other genus mines are also not getting much friendliness (ie it does not seem to be limited to glycinemine), but LegumeMine is getting most of the mines to be friendly with it (except GlycineMine, though possibly the unfriendliness is sporadic, I have seen it return results to LegumeMine).

**Comment by svengato** [comment]
These production mines are missing that `friendlymines.linkGenerator=` line: ArachisMine, LupinusMine.

**Comment by svengato** [comment]
This does not affect GlycineMine, but some of the mines did not update their friendly mines from common names to genus names.

**Comment by adf-ncgr** [comment]
yes, I noticed that too; we should fix it, though it doesn't seem to be the only problem (I think I may have already committed but not yet pushed an update on GlycineMine to address a couple that it didn't have right)

**Comment by svengato** [comment]
GlycineMine's `webapp/src/main/web.properties` has a swap file, are you in the middle of editing it?

**Comment by svengato** [comment]
Also, should a mine include itself as a friendly mine? It looks like some do and some do not.

**Comment by adf-ncgr** [comment]
> GlycineMine's `webapp/src/main/web.properties` has a swap file, are you in the middle of editing it?

no, but I did have it open.

**Comment by adf-ncgr** [comment]
> Also, should a mine include itself as a friendly mine? It looks like some do and some do not.

I'm not sure. Some mines being friendly to themselves might make sense but others (legumemine) would likely be unhelpful. For now, let's just leave as is.

**Comment by adf-ncgr** [comment]
I'm starting to get the impression that the issue may be one caused by a data loading change. It seems like the current implementation can only find friends if it itself knows about their genes. This is true of legumemine and possibly was true of the other mines (e.g. Arachismine might have previously had shallow knowledge of what genes from other genera were present in the gene family, but it does not now). I suspect this may have been a change that Sam made without noticing the consequences. This is still very much a theory, but would seem to explain the asymmetric behavior.

**Comment by svengato** [comment]
I can look through Sam's change history, though mixed in with other tasks so as to avoid InterMine stress syndrome.
